### PR TITLE
[MI-2362]:Fixed review fixes of confluence PR 89 on mattermost

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**

### CODE REVIEW

File: server/mock.go [hunk 1]
```diff
@@ -11,3 +29,26 @@ func mockAnythingOfTypeBatch(argType string, numCalls int) []interface{} {
...
+
+func getValidUserAndChannelID() (string, string) {
+	return validUserID, validChannelID
+}
+
+func getMockUser(isAdmin bool) *model.User {
+	if isAdmin {
+		return &model.User{Id: "123", Roles: "system_admin"}
+	}
+	return &model.User{Id: "123", Roles: "system_user"}
```
Comment (ayusht2810) [inline server/mock.go]:
```suggestion
	}
	
	return &model.User{Id: "123", Roles: "system_user"}
```

File: server/mock.go [hunk 2]
```diff
@@ -11,3 +29,26 @@ func mockAnythingOfTypeBatch(argType string, numCalls int) []interface{} {
...
+	return mockAPI, &p
+}
+
+func getValidUserAndChannelID() (string, string) {
+	return validUserID, validChannelID
+}
+
+func getMockUser(isAdmin bool) *model.User {
+	if isAdmin {
+		return &model.User{Id: "123", Roles: "system_admin"}
```
Comment (ayusht2810) [inline server/mock.go]:
```suggestion
		return &model.User{Id: "123", Roles: model.SYSTEM_ADMIN_ROLE_ID}
```

File: server/mock.go (hunk 1 above)
Comment (ayusht2810) [inline server/mock.go]:
```suggestion
	return &model.User{Id: "123", Roles: SYSTEM_USER_ROLE_ID}
```

File: server/serializer/channel_subscription_test.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+		"space subscription": {
+			subscription: &SpaceSubscription{
+				SpaceKey: "TS",
+				BaseSubscription: BaseSubscription{
+					Alias:     "test",
+					BaseURL:   "https://test.com",
+					ChannelID: "testChannelID",
+					Events:    []string{CommentRemovedEvent, CommentUpdatedEvent},
+				},
+			},
```
Comment (ayusht2810) [inline server/serializer/channel_subscription_test.go]: we can have util to get space and page subscription, as they have same values for all test cases

**Review by ayusht2810 — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Pull request merged**